src/main/java/com/atlassian/db/replica/api/AuroraPostgresLsnReplicaConsistency.java

```diff
@@ -0,0 +1,99 @@
+package com.atlassian.db.replica.api;
+
+import com.atlassian.db.replica.internal.MonotonicMemoryCache;
+import com.atlassian.db.replica.spi.Cache;
+import com.atlassian.db.replica.spi.ReplicaConsistency;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+/**
+ * It remembers last write LSN (log sequence number) and compares it with LSN for subsequent reads to detect inconsistencies.
+ * If it cannot remember the LSN of last write, pessimistically assumes it's going to be inconsistent.
+ *
+ * @see <a href="https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-global-database-monitoring.html">Monitoring Aurora PostgreSQL-based Aurora global databases</a>
+ * <p>
+ * It's an PostgreSQL-based Aurora database engine specific implementation.
+ */
+public final class AuroraPostgresLsnReplicaConsistency implements ReplicaConsistency {
```

Comment:
AuroraPostgres -> Aurora?

Reply:
There's a [resolved convo about this](https://github.com/atlassian-labs/db-replica/pull/62#discussion_r595922980).